### PR TITLE
Bud 17 report user

### PR DESCRIPTION
**Pull request by lukemeierj**
### In this PR
* Report activities with full text area
* Report users with report modal
* When you report someone/something, you go back to your root view controller
* Reports persist to firebase

##### UI change
Screen Shot 2019-04-01 at 10 26 49 AM

image

### TODO
- [ ] Write tests 
- [ ] Rebase on master after #120 and #116 are done
- [ ] Drop both Pod commits and re-reinstall the pod to avoid ugly conflicts with master

### To test

* Report a user
    * Make sure a `user_report` document is created with correct IDs
    * Make sure recipient has blocker in `blocked_by` list
    * Make sure instigator has blocked in `blocked_users`
    * Make sure you return back to the root of your tab VC (i.e. discover/my activities/topics)
* Report an activity
    * Make sure a `activity_report` document is created with correct IDs
    * Make sure your `account` has the activity in its `blocked_activitites` array
    * Make sure you return back to root VC

### CODE REVIEW

**Review by noahtallen — APPROVED** [review]
This works amazingly. Really good work!

File: Buddies/Activities/View Activity/ViewActivityController.swift [hunk 1]
```diff
@@ -66,11 +66,17 @@ class ViewActivityController: UIViewController {
     }
 
     @objc func onReportTap(_ sender: Any) {
-        guard self.curMemberStatus != .owner else { return }
-        showCancelableAlert(withMsg: "What's wrong with this activity?", withTitle: "Report Activity", withAction: "Report", showTextEntry: true) { (didConfirm, msg) in
-            guard didConfirm, let activity = self.curActivity else { return }
-            FirestoreManager.reportActivity(activity.activityId, reportMessage: msg ?? "")
-        }
+        guard self.curMemberStatus != .owner,
+            let parentNav = BuddiesStoryboard.Report.viewController() as? UINavigationController,
```
Comment (noahtallen) [inline Buddies/Activities/View Activity/ViewActivityController.swift:70]: I'm a little confused by the naming convention here - how is report the parent nav if we're presenting it?

File: Buddies/Controls/ReportModalVC.swift [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    var isUser: Bool { get { return userId != nil }}
+    
+    @IBOutlet weak var warningText: UILabel!
+    
+    var pronoun: String {
+        get { return isUser ? "them" : "it" }
+    }
+    
+    var type: String {
+        get { return isUser ? "user" : "activity" }
```
Comment (noahtallen) [inline Buddies/Controls/ReportModalVC.swift:28]: I like what you did with the getters!

File: Buddies/Onboarding/SignUpInfoVC.swift [hunk 3]
```diff
@@ -66,11 +66,11 @@ class SignUpInfoVC: LoginBase, UIImagePickerControllerDelegate, UINavigationCont
         bioText.delegate = self
         bioText.textColor = UIColor.lightGray
         bioText.layer.cornerRadius = Theme.cornerRadius
-        bioText.textContainerInset = UIEdgeInsets(top: 12 ,left: 7, bottom: 12, right: 7)
+        bioText.textContainerInset = Theme.textAreaInset
```
Comment (noahtallen) [inline Buddies/Onboarding/SignUpInfoVC.swift:69]: nice

File: Buddies/Controls/Theme.swift [hunk 4]
```diff
@@ -16,6 +16,8 @@ class Theme {
     static let themeAlt =  UIColor(red: 126/255, green: 92/255, blue: 171/255, alpha: 1.0)
     static let white = UIColor(red: 1, green: 1, blue: 1, alpha: 0.8)
     static let bad = UIColor.red
+    static let textAreaInset = UIEdgeInsets(top: 12 ,left: 7, bottom: 12, right: 7)
```
Comment (noahtallen) [inline Buddies/Controls/Theme.swift:19]: i had made the `// Other constants` part of this for things like this, and then just had colors in this top section. obviously not a big deal whatsoever

Comment (lukemeierj) [reply Buddies/Controls/Theme.swift:19]: Nice.  I committed this before I rebased, so probably didn't see it.

**Pull request closed without merging**